Implement inserting correct tenant in poLine pieces locations using location and holding ids

azizbekxm opened the pull request.

SerhiiNosko commented on src/main/resources/application.yml:
```diff
@@ -0,0 +1,24 @@
+    url: ${KONG_URL}
+  api:
+    url: ${FOLIO_URL}
+    tenant: consortium
```
it can be different tenantId for central tenant, so should be as env variable

azizbekxm replied:
changed to env variables

SerhiiNosko commented on src/main/java/org/folio/acquisitions_tools/service/LoginService.java:
```diff
@@ -0,0 +1,68 @@
+  private String password;
+
+  public String getToken() {
+    if (token == null) {
```
better to not have statful fields, better to login each time when doing rest call to avoid potential errors

azizbekxm replied:
refactored to use as method

SerhiiNosko commented on src/main/java/org/folio/acquisitions_tools/service/DatabaseService.java:
```diff
@@ -0,0 +1,120 @@
+             )
+        \s""";
+
+      jdbcTemplate.update(String.format(sql, tenant, locationTenantId, locationId));
```
would be  good to add logs that particular location is processed. Also update() method returns count of records that was updated, it is useful info for log as well

azizbekxm replied:
added logs

SerhiiNosko commented on .gitattributes:
```diff
@@ -0,0 +1,2 @@
+/mvnw text eol=lf
+*.cmd text eol=crlf
```
Readme file also should be added that will be used by Ops with information how to use this tool

azizbekxm commented:
Tested on rancher, working as expected, here is logs from it
```
2025-04-09T17:21:16.532+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.LocationService            : updatePoLineLocations:: Updating poLine locations by locationId
2025-04-09T17:21:17.921+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '0' Updated PoLines for location: 184aae84-a5bf-4c6a-85ba-4a7c73026cd5=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:18.221+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '1' Updated PoLines for location: fcd64ce1-6995-48f0-840e-89ffa2288371=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:18.529+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '0' Updated PoLines for location: 758258bc-ecc1-41b8-abca-f7b610822ffd=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:18.835+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '0' Updated PoLines for location: f34d27c6-a8eb-461b-acd6-5dea81771e70=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:19.068+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '1' Updated PoLines for location: 9d1b77e8-f02e-4b7f-b296-3f2042ddac54=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:19.296+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '0' Updated PoLines for location: 53cf956f-c1df-410b-8bea-27f712cca7c0=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:19.526+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByLocation:: '0' Updated PoLines for location: b241764c-1466-4e1d-a028-1a3684a5da87=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:21:25.255+05:00  WARN 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.LocationService            : Found 1 holdings with duplicate IDs:
2025-04-09T17:21:25.256+05:00  WARN 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.LocationService            : Holding ID: 10cd3a5a-d36f-4c7a-bc4f-e1ae3cf820c9 appears in tenants: university, college, consortium
2025-04-09T17:21:25.257+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.LocationService            : Removed 1 duplicate holdings, returning 22 unique holdings
2025-04-09T17:21:26.309+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.LocationService            : updatePoLineLocations:: Updating poLine locations by holdingId
2025-04-09T17:21:27.844+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '1' Updated PoLines for holding: 5e0f3f12-79a7-4167-8ba9-6d5325904670=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:28.153+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 0c45bb50-7c9b-48b0-86eb-178a494e25fe=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:28.385+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: c4a15834-0184-4a6f-9c0c-0ca5bad8286d=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:28.613+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: e6d7e91a-4dbc-4a70-9b38-e000d2fbdc79=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:28.862+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 247f1832-88be-4a84-9638-605ffde308b3=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:29.089+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 7fd55f10-6aec-4e25-a3cd-9ac7412ca26a=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:29.318+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 7a2443bc-fe66-40d5-9400-9a800fdf013e=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:29.547+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 7af9a14d-6e92-4827-acab-eba65e08be6c=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:29.780+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '1' Updated PoLines for holding: 6f5c95d3-f445-455a-b545-84eefc956ed6=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:30.009+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 55f48dc6-efa7-4cfe-bc7c-4786efe493e3=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:30.302+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 67cd0046-e4f1-4e4f-9024-adf0b0039d09=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:30.613+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 68872d8a-bf16-420b-829f-206da38f6c10=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:30.917+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 9e8dc8ce-68f3-4e75-8479-d548ce521157=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:31.227+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 65032151-39a5-4cef-8810-5350eb316300=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:31.533+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: e3ff6133-b9a2-4d4c-a1c9-dc1867d4df19=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:31.840+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 704ea4ec-456c-4740-852b-0814d59f7d21=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:32.067+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 13767c78-f8d0-425e-801d-cc5bd475856a=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:32.295+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 65cb2bf0-d4c2-4886-8ad0-b76f1ba75d61=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:32.527+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 133a7916-f05e-4df4-8f7f-09eb2a7076d1=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:32.760+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: e9285a1c-1dfc-4380-868c-e74073003f43=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:33.074+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: 1ab1f67e-6ab8-40f0-8cc1-a199db309070=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:21:33.303+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-1] o.f.a.service.DatabaseService            : updatePoLineLocationsByHolding:: '0' Updated PoLines for holding: fb7b70f1-b898-4924-a991-0e4b6312bb5f=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:22.946+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.LocationService            : updatePieces:: Updating pieces locations by locationId
2025-04-09T17:22:23.983+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: 184aae84-a5bf-4c6a-85ba-4a7c73026cd5=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:24.214+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: fcd64ce1-6995-48f0-840e-89ffa2288371=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:24.442+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: 758258bc-ecc1-41b8-abca-f7b610822ffd=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:24.668+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: f34d27c6-a8eb-461b-acd6-5dea81771e70=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:24.904+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: 9d1b77e8-f02e-4b7f-b296-3f2042ddac54=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:25.189+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: 53cf956f-c1df-410b-8bea-27f712cca7c0=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:25.416+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByLocation:: '0' Updated Pieces for location: b241764c-1466-4e1d-a028-1a3684a5da87=consortium and locationTenant: consortium in tenant: consortium
2025-04-09T17:22:28.488+05:00  WARN 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.LocationService            : Found 1 holdings with duplicate IDs:
2025-04-09T17:22:28.489+05:00  WARN 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.LocationService            : Holding ID: 10cd3a5a-d36f-4c7a-bc4f-e1ae3cf820c9 appears in tenants: university, college, consortium
2025-04-09T17:22:28.489+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.LocationService            : Removed 1 duplicate holdings, returning 22 unique holdings
2025-04-09T17:22:29.085+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.LocationService            : updatePieces:: Updating pieces locations by holdingId
2025-04-09T17:22:30.412+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '1' Updated Pieces for holding: 5e0f3f12-79a7-4167-8ba9-6d5325904670=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:30.639+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 0c45bb50-7c9b-48b0-86eb-178a494e25fe=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:30.866+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: c4a15834-0184-4a6f-9c0c-0ca5bad8286d=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:31.126+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: e6d7e91a-4dbc-4a70-9b38-e000d2fbdc79=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:31.434+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 247f1832-88be-4a84-9638-605ffde308b3=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:31.663+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 7fd55f10-6aec-4e25-a3cd-9ac7412ca26a=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:31.946+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 7a2443bc-fe66-40d5-9400-9a800fdf013e=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:32.185+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 7af9a14d-6e92-4827-acab-eba65e08be6c=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:32.414+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '1' Updated Pieces for holding: 6f5c95d3-f445-455a-b545-84eefc956ed6=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:32.641+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 55f48dc6-efa7-4cfe-bc7c-4786efe493e3=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:32.867+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 67cd0046-e4f1-4e4f-9024-adf0b0039d09=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:33.174+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 68872d8a-bf16-420b-829f-206da38f6c10=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:33.401+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 9e8dc8ce-68f3-4e75-8479-d548ce521157=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:33.626+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 65032151-39a5-4cef-8810-5350eb316300=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:33.894+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: e3ff6133-b9a2-4d4c-a1c9-dc1867d4df19=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:34.122+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 704ea4ec-456c-4740-852b-0814d59f7d21=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:34.352+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 13767c78-f8d0-425e-801d-cc5bd475856a=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:34.608+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 65cb2bf0-d4c2-4886-8ad0-b76f1ba75d61=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:34.921+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 133a7916-f05e-4df4-8f7f-09eb2a7076d1=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:35.223+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: e9285a1c-1dfc-4380-868c-e74073003f43=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:35.530+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: 1ab1f67e-6ab8-40f0-8cc1-a199db309070=consortium and holdingTenant: consortium in tenant: consortium
2025-04-09T17:22:35.838+05:00  INFO 6440 --- [acquisitions-tools] [nio-8080-exec-4] o.f.a.service.DatabaseService            : updatePiecesByHoldings:: '0' Updated Pieces for holding: fb7b70f1-b898-4924-a991-0e4b6312bb5f=consortium and holdingTenant: consortium in tenant: consortium

```

SerhiiNosko commented on README.md:
````diff
@@ -0,0 +1,52 @@
+  curl -X POST http://localhost:8080/api/locations/update-pieces
+  ```
+- **Success Response:** `200 OK` with body `"Pieces update completed successfully"`
+- **Error Response:** `500 Internal Server Error` with details
````
Can you please add info how to run this without IDE, I see mvnw is added, so ops even do not need to install maven that is good

BKadirkhodjaev commented on README.md:
```diff
@@ -0,0 +1,52 @@
+- `FOLIO_USERNAME`: FOLIO API username
+- `FOLIO_PASSWORD`: FOLIO API password
+
+The application runs on port `8080` by default.
```
Can be port be changed? They may have some apps already running on 8080

BKadirkhodjaev replied:
Maybe some VM flag (.e.g. -Dserver.port=29901) in your `java -jar ..` command can be added to help override the port

BKadirkhodjaev commented on src/main/java/org/folio/acquisitions_tools/service/DatabaseService.java:
```diff
@@ -0,0 +1,133 @@
+              (
+                SELECT jsonb_agg(
+                  CASE
+                    WHEN location->>'locationId' = '%s' AND NOT (location ? 'tenantId') THEN\s
```
Is this "\s" needed here at "THEN\s"? 

Maybe some copy-pasting left this

azizbekxm replied:
removed

azizbekxm replied:
apparently it is seems required, can't run application without it, but I removed other places

BKadirkhodjaev commented on src/main/java/org/folio/acquisitions_tools/controller/LocationController.java:
```diff
@@ -0,0 +1,36 @@
+@RequestMapping("/api/locations")
+public class LocationController {
+
+  @Autowired
```
Maybe better to replace @Autowired with @RequiredArgsConstructor on the class level and making the member field private final, e.g. `private final LocationService locationService;`

BKadirkhodjaev commented on src/test/java/org/folio/acquisitions_tools/AcquisitionsToolsApplicationTests.java:
```diff
@@ -0,0 +1,13 @@
+@SpringBootTest
+class AcquisitionsToolsApplicationTests {
+
+	@Test
```
Maybe need to re-indent it to 2 spaces

BKadirkhodjaev commented on src/main/java/org/folio/acquisitions_tools/service/LoginService.java:
```diff
@@ -0,0 +1,57 @@
+
+@Service
+public class LoginService {
+  private final RestTemplate restTemplate = new RestTemplate();
```
Better to move RestTemplate into a @Configuration-annotated class and make it a Bean

azizbekxm replied:
Moved

BKadirkhodjaev commented on src/main/java/org/folio/acquisitions_tools/service/DatabaseService.java:
```diff
@@ -0,0 +1,133 @@
+           WHERE\s
+               jsonb->>'holdingId' = '%s'
+               AND NOT (jsonb ? 'receivingTenantId')
+          \s""";
```
Also here "\s"

BKadirkhodjaev commented on src/main/java/org/folio/acquisitions_tools/service/DatabaseService.java:
```diff
@@ -0,0 +1,133 @@
+           WHERE\s
+               jsonb->>'locationId' = '%s'
+               AND NOT (jsonb ? 'receivingTenantId')
+          \s""";
```
Also here "\s"

BKadirkhodjaev commented on pom.xml:
```diff
@@ -0,0 +1,88 @@
+	</parent>
+	<groupId>org.folio</groupId>
+	<artifactId>acquisitions-tools</artifactId>
+	<version>0.0.1-SNAPSHOT</version>
```
Pom.xml possibly needs to be re-indented to 2 spaces

azizbekxm commented on README.md:
````diff
@@ -0,0 +1,111 @@
+   Navigate to the project root directory and run the following Maven command to build the executable JAR:
+
+   ```bash
+   mvn package
````
update to  ./mvnw clean install as far as I know it not required mvn to install

SerhiiNosko commented:
@azizbekxm thanks for this quick and great work for this tool
